File: src/ai/VectorStores/DotNet/ChromaVectorStoreOptions.cs
namespace VectorStores.DotNet;

public sealed class ChromaVectorStoreOptions
{
    public string BaseUrl { get; set; } = "http://localhost:8001";

    public string Tenant { get; set; } = "default_tenant";

    public string Database { get; set; } = "default_database";

    public string CollectionName { get; set; } = "recipe_embeddings";

    public TimeSpan Timeout { get; set; } = TimeSpan.FromSeconds(30);
}

File: src/ai/VectorStores/NestJS/chroma-vector-store-options.ts
export type ChromaVectorStoreOptions = {
  baseUrl: string;
  tenant: string;
  database: string;
  collectionName: string;
  timeoutMs: number;
};

export const CHROMA_VECTOR_STORE_OPTIONS = Symbol('CHROMA_VECTOR_STORE_OPTIONS');

export function createChromaVectorStoreOptions(): ChromaVectorStoreOptions {
  return {
    baseUrl: (process.env.CHROMA_VECTOR_STORE_BASE_URL || 'http://localhost:8001').replace(/\/+$/, ''),
    tenant: process.env.CHROMA_VECTOR_STORE_TENANT || 'default_tenant',
    database: process.env.CHROMA_VECTOR_STORE_DATABASE || 'default_database',
    collectionName: process.env.CHROMA_VECTOR_STORE_COLLECTION_NAME || 'recipe_embeddings',
    timeoutMs: Number(process.env.CHROMA_VECTOR_STORE_TIMEOUT_MS || 30000),
  };
}